### ISSUE TITLE
fancyref.cwl found in another repo

### ISSUE THREAD

**Issue by DanteSung** [texstudio-org/texstudio#844]
### Environment
- TeXstudio: 2.12.16
- OS:  Mac
- TeX distribution:  texlive
Hey! 

Lately, I found the  `fancyref` package super useful when writing some reports but I can't find the corresponding autocompletion in TXS.

I found the `cwl` file for the `fancyref` package is missing in the built-in folder, so I googled it and found the file in another [git repo](https://github.com/LaTeXing/LaTeX-cwl/blob/master/fancyref.cwl).

I think this would be helpful for the auto-completion of the `fancyref` package.

Peace

**Comment by sunderme** [comment]
you could place the file in ~/.config/texstudio/completion/user and there txs will find it.

**Comment by DanteSung** [comment]
> you could place the file in ~/.config/texstudio/completion/user and there txs will find it.

Thanks! I've put it in the folder according to the official guide. Just want to make sure if it's possible to include some more `cwl` files in a future release?

Also, I put `#r` after `\fref{}` to show the label list in TXS, but it comes along with the comletion of `\fref`. Is it possible to show the label list in the auto-completion like `\ref{}`?

**Comment by sunderme** [comment]
sure we can add more cwls, best done with pull requests, though file attachments are fine,too.
We would prefer cwls which are adapted for txs (like the #r etc)

> Also, I put #r after \fref{} to show the label list in TXS, but it comes along with the comletion of \fref

I don't understand the issue here ?

**Comment by DanteSung** [comment]
Ah, sorry about the ambiguity.
It's about the content in the `cwl` file. In their [repo](https://github.com/LaTeXing/LaTeX-cwl/blob/master/fancyref.cwl), the completions are defined as
```
# fancyref.sty
# Michael S. 2010/11/20
#
# real commands, macros not included
\fref{label}
\Fref{label}
.....
```
After being added to `~/.config/texstudio/completion/user`, the auto-completion for those two commands show up. However, since it's a reference command, I want it to show my label list as well (containing all the defined labels like fig:xxx, eq:xxx). To achieve this aim, I change the command from `\fref{label}` to `\fref{label} #r`. After this, the labels comes with `\fref{}` in the waitlist of auto-completion like
```
\fref{label1}
\fref{label2}
...
```
Though I can work with it, I'm still wondering how I can modify this file to make it work like `\ref{}`. For `\ref`, after typing in the command, when I type `{`, the label list will pop up in the auto-completion like
```
{label1}
{label2}
...
```
How can I make `\fref{}` work like that? Thanks!

**Comment by muzimuzhi** [comment]
@DanteSung 
```
\fref{label}#r
\Fref{label}#r
```
works alright. By the way, I have prepared a more comprehensive `fancyref.cwl` file and you can check it in  push request #846 .

**Comment by timhoffm** [comment]
Closed in #864.
